core/src/main/java/org/infinispan/commands/tx/PrepareCommand.java

```diff
@@ -211,7 +211,7 @@ public void readFrom(ObjectInput input) throws IOException, ClassNotFoundExcepti
       super.readFrom(input);
       onePhaseCommit = input.readBoolean();
       retriedCommand = input.readBoolean();
-      modifications = MarshallUtil.unmarshallArray(input, WriteCommand[]::new);
+      modifications = MarshallUtil.unmarshallArray(input, WriteCommand::array);
```

Comment:
Are prepare commands with 0 modifications that common?

Reply:
Apparently not, otherwise we would be hitting the ClassCastException more often. I could remove that as premature optimization; I have just replaced all usages.